src/Microsoft.AspNetCore.Authentication.Google/GoogleHandler.cs

```diff
@@ -32,7 +32,10 @@ public GoogleHandler(HttpClient httpClient)
             request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokens.AccessToken);
 
             var response = await Backchannel.SendAsync(request, Context.RequestAborted);
-            response.EnsureSuccessStatusCode();
+            if (!response.IsSuccessStatusCode)
+            {
+                throw new HttpRequestException($"An error occurred when retrieving user information ({response.StatusCode}). Please check if the authentication information is correct and the corresponding Google API is enabled.");
```

Comment:
IMHO, this message is not precise enough. AFAIK, it's actually the "Google+ API" that must be enabled.